extensions/wikia/Recirculation/js/helpers/LateralHelper.js

```diff
@@ -4,29 +4,35 @@ define('ext.wikia.recirculation.helpers.lateral', [
 	'wikia.window',
 	'wikia.abTest',
 ], function ($, w, abTest) {
-	var libraryLoaded = false;
+	var libraryLoaded = false,
+		queue = [];
 
 	function loadLateral(callback) {
-		if (libraryLoaded) {
-			if (callback) {
-				callback(w.lateral);
-			}
-			return;
+		queue.push(callback);
+
+		if (w.lateral) {
+			processQueue(w.lateral);
 		}
+		if (!libraryLoaded) {
+			var lateralScript = document.createElement('script');
+			libraryLoaded = true;
 
-		var lateralScript = document.createElement('script');
+			lateralScript.src = 'https://assets.lateral.io/recommendations.js';
+			document.getElementsByTagName('body')[0].appendChild(lateralScript);
 
-		lateralScript.src = 'https://assets.lateral.io/recommendations.js';
-		document.getElementsByTagName('body')[0].appendChild(lateralScript);
+			// This function is called when the Lateral script has loaded
+			w.onLoadLateral = function(lateral) {
+				w.lateral = lateral;
+				processQueue(lateral);
+			}
+		}
+	}
 
-		// This function is called when the Lateral script has loaded
-		w.onLoadLateral = function(lateral) {
-			w.lateral = lateral;
-			libraryLoaded = true;
+	function processQueue(lateral) {
+		while(queue.length > 0) {
+			var callback = queue.pop();
```

Comment:
`pop` will start with the item that was `push` most recently. `shift` should be better.